### PR TITLE
Fix up tests after adding KDD container ID support.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Remove some special case logic from the tests now that KDD round-trips the container ID (https://github.com/projectcalico/libcalico-go/pull/1472).

Related issue:
https://github.com/projectcalico/calico/issues/4710

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Review by caseydavenport — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
